dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -935,37 +1046,28 @@ def filter_cells_by_outliers(
-            Maximal number of genes with expression for a cell in the data from the unspliced layer.
-        max_expr_genes_p: `float` (default: `np.inf`)
-            Maximal number of protein with expression for a cell in the data from the protein layer.
-        shared_count: `int` or `None` (default: `None`)
-            The minimal shared number of counts for each cell across genes between layers.
-
-    Returns
-    -------
-        adata: :class:`~anndata.AnnData`
-            An updated AnnData object with use_for_pca as a new column in obs to indicate the selection of cells for
-            downstream analysis. adata will be subsetted with only the cells pass filtering if keep_filtered is set to
-            be False.
+    Args:
+        adata (anndata.AnnData): an AnnData object.
+        filter_bool (Union[np.ndarray, None], optional): a boolean array from the user to select cells for downstream analysis. Defaults to None.
+        layer (str, optional): the layer (include X) used for feature selection. Defaults to "all".
+        keep_filtered (bool, optional): whether to keep cells that don't pass the filtering in the adata object. Defaults to False.
+        min_expr_genes_s (int, optional): minimal number of genes with expression for a cell in the data from the spliced layer (also used for X). Defaults to 50.
+        min_expr_genes_u (int, optional): minimal number of genes with expression for a cell in the data from the unspliced layer. Defaults to 25.
+        min_expr_genes_p (int, optional): minimal number of genes with expression for a cell in the data from in the protein layer. Defaults to 1.
+        max_expr_genes_s (float, optional): maximal number of genes with expression for a cell in the data from the spliced layer (also used for X). Defaults to np.inf.
+        max_expr_genes_u (float, optional): maximal number of genes with expression for a cell in the data from the unspliced layer. Defaults to np.inf.
+        max_expr_genes_p (float, optional): maximal number of protein with expression for a cell in the data from the protein layer. Defaults to np.inf.
+        shared_count (Union[int, None], optional): the minimal shared number of counts for each cell across genes between layers. Defaults to None.
+        spliced_key (str, optional): name of the layer storing spliced data. Defaults to "spliced".
+        unspliced_key (str, optional): name of the layer storing unspliced data. Defaults to "unspliced".
+        protein_key (str, optional): name of the layer storing protein data. Defaults to "protein".
+        obs_store_key (str, optional): name of the layer to store the filtered data. Defaults to "pass_basic_filter".
+
+    Raises:
+        ValueError: the layer provided is invalid.
+
+    Returns:
+        anndata.AnnData: An updated AnnData object indicating the selection of cells for downstream analysis. adata will be subsetted with only the cells pass filtering if keep_filtered is set to be False.
```

Comment:
adata: An updated AnnData object indicating the selection of cells for downstream analysis. adata will be subsetted with only the cells pass filtering if keep_filtered is set to be False.